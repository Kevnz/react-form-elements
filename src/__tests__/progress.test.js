import React from 'react'
import { render, cleanup } from 'react-testing-library'

import Progress from '../progress'

describe('Progress component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { getByText, container } = render(
        <Progress label="My Progress Element" initialValue={50} />
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('My Progress Element')
      expect(label.textContent).toBe('My Progress Element')
    })
  })
  describe('handling input', () => {
    it('should have a value set', () => {
      const { container } = render(
        <Progress label="My Example Progress" initialValue={20} max={100} />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = container.querySelector('progress')

      expect(input.getAttribute('value')).toBe('20')
    })
  })
})
