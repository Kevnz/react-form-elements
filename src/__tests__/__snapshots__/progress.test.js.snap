// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row rfe-progress"
>
  <label
    for="id-1-1"
  >
    My Progress Element
  </label>
  <progress
    id="id-1-1"
    name="Progress"
    value="50"
  />
</div>
`;

exports[`Progress component handling input should have a value set 1`] = `
<div
  class="rfe-form__row rfe-progress"
>
  <label
    for="id-2-1"
  >
    My Example Progress
  </label>
  <progress
    id="id-2-1"
    max="100"
    name="Progress"
    value="20"
  />
</div>
`;
